data/spells/scripts/support/levitate.lua

```diff
@@ -0,0 +1,36 @@
+function onCastSpell(creature, variant)
+	local currentPosition = creature:getPosition()
+	local destinationPosition = creature:getPosition()
+	destinationPosition:getNextPosition(creature:getDirection())
```

Comment:
But the function ``Position:getNextPosition`` modifies the caller's position ``self`` fields, instead of returning them. So I do what you did then ``currentPosition`` will be modified and instead will hold the next position.

We could prevent that if:

```lua
function Position:getNextPosition(direction, steps)
	local offset = Position.directionOffset[direction]
	if offset then
		steps = steps or 1
		self.x = self.x + offset.x * steps
		self.y = self.y + offset.y * steps
	end
end
```

should be modified to return them, instead of modifying **self** fields.

```lua
function Position:getNextPosition(direction, steps)
	local offset = Position.directionOffset[direction]
	return offset and Position(self.x + offset.x * steps or 1, self.y + offset.y * steps, self.z)
end
```